scalar-decomposition/README.md

```diff
@@ -0,0 +1,124 @@
+# Scalar decomposition for point-multiplication on Elliptic Curves
+This algorithm provides a decomposition of a scalar in two scalars with half bit-length. The implementation of it helps to speed up the elliptic curve point multiplication. 
+
+## Motivation
+The idea of using this algorithm for accelerating point multiplication is given in [*Guide to Elliptic Curve Criptography*][link-libro]. 
+Given an Elliptic Curve *E*, let *&lambda;* be the root of the characteristic polynomial of an Endomorphism over *E*, and let *n* be the order of the Group of the curve. If *k* is a constant in in [0,...,n-1], then there exist *k<sub>1* and *k<sub>2* in [0,...,n-1] such that
+*k=k<sub>1</sub>+k<sub>2</sub>&lambda;* mod*n*. The components *k<sub>1* and *k<sub>2* are half bit-long than *k*, making the point multiplication in *E* faster, since for every *P* in *E*
```

Comment:
```suggestion
*k=k<sub>1</sub>+k<sub>2</sub>&lambda;* mod*n*. The components *k<sub>1* and *k<sub>2* are half bit-long than *k*, making the point multiplication in *E* faster, since for every *P* in *E* we have
```